cmd/snap/cmd_set.go

```diff
@@ -155,6 +161,48 @@ func (x *cmdSet) Execute([]string) error {
 	return nil
 }
 
+func validateJsonKeys(v interface{}) error {
```

Comment:
Please rename to `validateJSONKeys` or something of the sort -- `JSON` should be all-caps rather than camelCase.

Reply:
Also, this does more than just validate JSON keys, right? Should this just be called `validateJSON`?

Reply:
The JSON argument itself, it's already being validated in the sense that if it's parsed successfully into GO structures or not.  

My goal on this PR is to navigate through all nested maps and arrays, validating the `JSON keys` only, using this regex (through the `ValidateKey` function):

https://github.com/snapcore/snapd/blob/5c8d8431baa425464b279ff26b8c44eecb9aab22/overlord/configstate/config/helpers.go#L36

Because, AFAICT, the values for a config in the `snap set` command can be pretty much anything, right? Since they are user-defined.